### DISCUSSION TITLE
Problem with pyinstaller

### DISCUSSION THREAD

**Discussion by kwame-Owusu** [community/community#55943] [Programming Help] [answered]
### Body

I have made a GUI app with tkinter and customtkinter, and i want to make it into an .exe file, although I am able to do so, when I open the .exe file the app will open and close immediately, can I get any advice on how to solve this problem?, thanks in advance. the app can be found in my account in a repository called pomodoro(a gui app to help study with the pomodoro technique)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by KNJSoft** [comment]
Sure. PyInstaller is a Python module that can be used to create standalone executables from Python scripts. This can be useful for distributing your Python applications to users who do not have Python installed.

To use PyInstaller, you will first need to install it. You can do this using the following command:
Code snippet

pip install pyinstaller

Use code with caution. [Learn more](https://bard.google.com/faq#coding)

Once PyInstaller is installed, you can create an executable from your Python script using the following command:
Code snippet

pyinstaller <script_name>

Use code with caution. [Learn more](https://bard.google.com/faq#coding)

For example, to create an executable from the script myscript.py, you would use the following command:
Code snippet

pyinstaller myscript.py

Use code with caution. [Learn more](https://bard.google.com/faq#coding)

This will create a folder called dist in the same directory as your Python script. Inside the dist folder, you will find an executable file with the same name as your Python script.

You can run the executable file by double-clicking on it. The executable file will run your Python script and display the output on the screen.

PyInstaller supports a variety of options that can be used to customize the creation of your executable file. For more information on these options, please refer to the PyInstaller documentation.

Here are some of the most commonly used options:

    -F or --onefile: This option will create a single executable file that contains all of the code and data for your Python application.
    -w or --windowed: This option will create a windowed executable file. By default, PyInstaller will create a console-based executable file.
    -i or --icon: This option allows you to specify an icon for your executable file.

**Comment by Agestar88** [comment]
-F or --onefile: This option will create a single executable file that contains all of the code and data for your Python application.
-w or --windowed: This option will create a windowed executable file. By default, PyInstaller will create a console-based executable file.
-i or --icon: This option allows

**Comment by kwame-Owusu** [comment, marked as answer]
I found the answer, thanks for the suggestions, it seems that the library i was using customtkinter had some problems with the packaging, but I went to the documentatation and found the fix, thanks for the replies
